### PR TITLE
Improve documentation and examples

### PR DESCRIPTION
**Pull request by yadaiio**
This pull request fixes a few typos with the help of https://github.com/szepeviktor/typos-on-you. 

Builds on top of: https://github.com/clue/reactphp-sqlite/pull/43.

### CODE REVIEW

**Review by SimonFrings — APPROVED** [review]
Looks good to me :+1:

File: README.md [hunk 1]
```diff
@@ -31,9 +31,9 @@ tar stream into the `Decoder` which emits "entry" events for each individual fil
 
 require __DIR__ . '/vendor/autoload.php';
 
-$stream = new ReadableResourceStream(fopen('archive.tar', 'r'));
+$stream = new React/Stream/ReadableResourceStream(fopen('archive.tar', 'r'));
```
Comment (SimonFrings) [inline README.md]: You're currently using the wrong slashes in here, I added a suggested change below. My suggestions also aligns with PSR-4, maybe worth to take a look at https://www.php-fig.org/psr/psr-4/ 

```suggestion
$stream = new React\Stream\ReadableResourceStream(fopen('archive.tar', 'r'));
```

File: README.md [hunk 2]
```diff
@@ -31,9 +31,9 @@ tar stream into the `Decoder` which emits "entry" events for each individual fil
...
 
-$decoder = new Decoder();
+$decoder = new Clue/React/Tar/Decoder();
```
Comment (SimonFrings) [inline README.md]: You're currently using the wrong slashes in here, I added a suggested change below. My suggestions also aligns with PSR-4, maybe worth to take a look at https://www.php-fig.org/psr/psr-4/ 

```suggestion
$decoder = new Clue\React\Tar\Decoder();
```

**Review by SimonFrings — CHANGES_REQUESTED** [review]
Thanks for updating this PR, added a few remarks below :+1:

These changes look similar to https://github.com/clue/reactphp-sqlite/pull/43, maybe include a reference to this ticket in your description and update the title?

**Comment by yadaiio** [comment]
Thank you @SimonFrings for informing me about this. Changes should be okay by now, I also changed the title of this pull request and added the references. Thank you 👍

**Review by SimonFrings — APPROVED** [review]
Looks good to me :+1:

**Review by clue — APPROVED** [review]
@yadaiio Thanks for looking into this, changes LGTM! :shipit:

**Pull request merged**